test/core_tests.js

```diff
@@ -1,138 +1,156 @@
-//
-// For now this just contains a single transition in order to get QUnit committed.
-// Complete test coverage needs to be added later.
-QUnit.test( "Impress Core API - Work in progress", function( assert ) {
-  // impress.js itself uses event listeners to manipulate most CSS classes. 
-  // Wait a short while before checking, to avoid race. 
-  // (See assertStepEnterWrapper and assertStepLeaveWrapper.)
-  var wait = 5; // milliseconds
-
-  var done = assert.async();
-  var step1 = document.querySelector( "div#step-1" );
-  var step2 = document.querySelector( "div#step-2" );
-  var root  = document.querySelector( "div#impress" );
-
-  // Things to check on impress:stepenter event -----------------------------//
-  var assertStepEnter = function( event ) {
-    assert.equal( event.target, step2,
-                  event.target.id + " triggered impress:stepenter event." );
-    assert.ok( step2.classList.contains("present"),
-               event.target.id + " set present css class." );
-    assert.ok( !step2.classList.contains("future"),
-               event.target.id + " unset future css class." );
-    assert.ok( !step2.classList.contains("past"),
-               event.target.id + " unset past css class." );
-    done();
+(function() {
+  var registerEventListener = function( target, eventType, callback ) {
+    target.addEventListener( eventType, callback );
+    return function removeRegisteredEventListener() {
+      target.removeEventListener( eventType, callback );
+    };
   };
-  
-  root.addEventListener( "impress:stepenter",  function(event) { 
-                                                   setTimeout( function() { assertStepEnter( event ) }, wait );
-                                               } 
-  );
-
-  // Do no-op tests first, then trigger a transition. //
-
-  assert.strictEqual( impress().goto(document.querySelector( "div#impress" )),
-                    false,
-                    "goto() to a non-step element fails, as it should." );
-  assert.strictEqual( impress().goto(),
-                    false,
-                    "goto(<nothing>) fails, as it should." );
-
-  assert.ok( impress().next(),
-             "next() called and returns ok (1->2)" );
-  assert.ok( step2.classList.contains("active"),
-             step2.id + " set active css class." );
-  assert.ok( !step1.classList.contains("active"),
-             step1.id + " unset active css class." );      
-});
-
-
-// Cleanup
-QUnit.done(function( details ) {
-  // Impress.js will set the hash part of the url, we want to unset it when finished
-  // Otherwise a refresh of browser page would not start tests from step 1
-  window.location.hash = "";    
-  // Add back vertical scrollbar so we can read results if there were failures. 
-  document.body.style.overflow = 'auto';
-});
 
+  QUnit.begin(function() {
+    impress().init();
+  });
+
+  QUnit.module( "Initialization" );
+
+    QUnit.test( "Global Scope", function( assert ) {
+      assert.expect( 1 );
```

Comment:
I am using Qunit’s assert validation as an additional check that we are not removing important assertions and making the tests still pass.